src/AtomBeautify.ts

```diff
@@ -1,203 +1,334 @@
+  private textInEditor(editor: TextEditor) {
+    if (!!editor.getSelectedText()) {
+      return editor.getSelectedText();
+    } else {
+      return editor.getText();
     }
+  }
 
-    private beautify(editor: Atom.TextEditor, data: BeautifyData) {
-      return this.unibeautify.beautify(data).then((result) => {
-        editor.setText(result);
-      }).catch(error => {
-        this.showError(error);
-      });
-    }
+  // Calls Unibeautify's beautify method
+  private beautify(data: BeautifyData): Promise<string> {
+    logger.info("Beautifying with", { data });
+    return this.unibeautify.beautify(data).catch(error => {
+      this.showError(error);
+      return Promise.reject(error);
+    });
+  }
 
-    private beautifyFile() {
-      // TODO
-    }
+  // On save format service provider
+  public provideOnSaveCodeFormat() {
+    return {
+      // tslint:disable-next-line:no-empty
+      formatOnSave: (editor: TextEditor) => {},
+      priority: 1,
+      grammarScopes: [],
+    };
+  }
 
-    private beautifyDirectory() {
-      // TODO
-    }
+  // Manual trigger format service provider
+  public provideFileCodeFormat() {
+    return {
+      // tslint:disable-next-line:no-empty
+      formatEntireFile: (editor: TextEditor, range: Range) => {
+        const text = editor.getText();
+        return Promise.resolve({
+          formatted: `Testing purposes only\n${text}`,
+          newCursor: 0
+        });
+      },
+      priority: 1,
+      grammarScopes: ["source.ts"],
+    };
+  }
 
-    private debug() {
-      // TODO
-    }
+  // Range format service provider
+  public provideRangeCodeFormat() {
+    return {
+      // tslint:disable-next-line:no-empty
+      formatCode: (editor: TextEditor, range: Range) => {},
+      priority: 1,
+      grammarScopes: [],
+    };
+  }
 
-    private configFromSettings(language?: Language) {
-      const config = atom.config.get("atom-beautify");
-      return language ? _.get(config, language.name) : config;
-    }
+  private beautifyFile({ target }: { target: any }) {
+    logger.info(target.dataset.path);
+    const filePath = target.dataset.path;
+    return this.showError(new Error("Could not beautify file"));
+  }
 
-    private openSettings() {
-      atom.workspace.open("atom://config/packages/atom-beautify");
-    }
+  private beautifyDirectory() {
+    // TODO
+  }
 
-    // ===== Helpers =====
-    private getScrollTop(editor: any): any {
-      const view = atom.views.getView(editor);
-      return view && view.getScrollTop();
-    }
+  private debug() {
+    // TODO
+  }
 
-    private setScrollTop(editor: any, value: any): void {
-      const view = atom.views.getView(editor);
-      view.setScrollTop(value);
+  // Check if a language is configured to beautify on save
+  private isBeautifyOnSave(language: string): Boolean {
+    const languageConfig = this.baseConfig.languages;
+    if (languageConfig.hasOwnProperty(language)) {
```

Comment:
This can be simplified to:

```diff
- if (languageConfig.hasOwnProperty(language)) {
-     return Boolean(_.get(languageConfig, language).beautify_on_save);
- }
- return false;
+ return Boolean(_.get(languageConfig, [language, "beautify_on_save"]));
```

See https://lodash.com/docs/#get